qutip/tests/test_operators.py

```diff
@@ -160,18 +156,20 @@ def test_super_operator_creation(to_test):
     assert implicit == explicit
 
 
-def test_num():
-    "Number operator"
-    n5 = num(5)
-    assert_equal(
-        np.allclose(n5.full(),
-                    np.diag([0 + 0j, 1 + 0j, 2 + 0j, 3 + 0j, 4 + 0j])),
-        True)
+@pytest.mark.parametrize(["to_test", "factor", "phase"], [
+    (qutip.position, 1, 1),
+    (qutip.momentum, 1j, -1),
+])
+def test_bidiagonal(to_test, factor, phase):
+    N = 5
+    operator = to_test(N)
+    expected = (np.diag((np.arange(1, N) / 2)**0.5, k=-1) +
+                np.diag((np.arange(1, N) / 2)**0.5, k=1) * phase) * factor
+    np.testing.assert_allclose(operator.full(), expected)
```

Comment:
It may be clearer just to test `position` and `momentum` separately and explicitly - it took me a second to work out what was going on, and then it also raises the question why `squeeze` isn't also considered "bidiagonal".